.github/workflows/update-toc.yml

```diff
@@ -0,0 +1,20 @@
+name: Update Table of Contents
+
+on: [push]
```

Comment:
Let's run this only on pushes that make changes inside the `docs` directory. No need to do this in every commit.